cpp/frameProcessor/include/FrameProcessorPlugin.h

```diff
@@ -90,45 +91,32 @@ class FrameProcessorPlugin : public IFrameCallback, public OdinData::IVersionedO
    */
   void add_metadata(OdinData::IpcMessage& message, const ParamMetadata& metadata) const
   {
-    std::string str = "metadata/";
-    static std::string& str_static = str;
-    static OdinData::IpcMessage&& message_static = std::move(message);
-
-    str += metadata.path;
-
+    std::string str = get_name() + metadata.path + '/';
     message.set_param(str +  "/type", metadata.type);
     message.set_param(str +  "/access_mode", metadata.access_mode);
-    if(metadata.has_min)
+    if(metadata.has_min){
       message.set_param(str +  "/min", metadata.min);
-    if(metadata.has_max)
+    }
+    if(metadata.has_max){
       message.set_param(str +  "/max", metadata.max);
-
-    str +=  "/allowed_vals[]";
-
-    // static variables are initialized only once, so ensure to reassign them each function call here
-    str_static = str;
-    message_static  = std::move(message);
-
-    struct VariantVisitor : public boost::static_visitor<void> 
-    {
-      void operator() (JsonNullType) const{
-        return;
-      }
-      void operator() (const std::string& s) const {
-        message_static.set_param(str_static, s);
-      }
-
-      void operator() (int n) const{
-        message_static.set_param(str_static, n);
-      }
-    };
-
-    auto first = metadata.allowed_vals.begin();
-    auto end = metadata.allowed_vals.end();
-    for (; first != end; ++first)
-	    boost::apply_visitor(VariantVisitor(), *first);
-    // std::for_each(metadata.allowed_vals.begin(), metadata.allowed_vals.end(), boost::apply_visitor(VariantVisitor()));
-
+    }
+
+    str +=  "/allowed_values[]";
+
+    auto first = metadata.allowed_values.begin();
```

Comment:
```suggestion
    auto itr = metadata.allowed_values.begin();
```

It isn't first for the whole for loop.